> I'm confused, the tests that care about `SOURCE_DATE_EPOCH` should use [this fixture](https://github.com/di/calver/blob/e9c8017caf3e3aff214fbb581ee7f366de55f629/tests/test_integration.py#L34-L39) which overwrites `SOURCE_DATE_EPOCH` if it's set in the environment.
> 
> Can you share the test failure you're trying to fix with this?

test_reproducible_build doesn't fail, instead test_version_str and test_version_true do.

```
$ SOURCE_DATE_EPOCH=1 PYTHONPATH=src/ pytest -v -k test_version_str
tests/test_integration.py::test_version_str FAILED                      [100%]

================================== FAILURES ===================================
______________________________ test_version_str _______________________________

dist = <stub(metadata=<stub(version='Thu Jan  1 00:00:01 1970')>)>
keyword = <stub()>, ignore_pkginfo = None

    def test_version_str(dist, keyword, ignore_pkginfo):
        value = "%c"

        calver.integration.version(dist, keyword, value)

>       assert dist.metadata.version == datetime.datetime.utcnow().strftime(value)
E       AssertionError: assert 'Thu Jan  1 00:00:01 1970' == 'Tue Apr 15 15:04:10 2025'
E
E         - Tue Apr 15 15:04:10 2025
E         + Thu Jan  1 00:00:01 1970

tests/test_integration.py:64: AssertionError
```

not only test_reproducible_build() cares about the epoch, but also every test that doesn't rely on PKG-INFO and should succeed. `SOURCE_DATE_EPOCH` has a priority higher than current date[1], so if the variable is set, the output won't match our expectation, which is always constructed from current date.

[1]: https://github.com/di/calver/blob/e9c8017caf3e3aff214fbb581ee7f366de55f629/src/calver/integration.py#L35

--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,3 +1,4 @@
+recursive-include tests *.py
 include pyproject.toml
 include tox.ini
 include *.md

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,0 +1,39 @@
+import os
+
+import pretend
+import pytest
+
+import calver.integration
+
+
+@pytest.fixture
+def original_version():
+    return pretend.stub()
+
+
+@pytest.fixture
+def dist(original_version):
+    return pretend.stub(metadata=pretend.stub(version=original_version))
+
+
+@pytest.fixture
+def keyword():
+    return pretend.stub()
+
+
+@pytest.fixture
+def ignore_pkginfo(monkeypatch):
+    # Ensure that the test doesn't unintently prefer to read from a PKG_INFO
+    # that might exist in the source directory, e.g. when testing a sdist
+    # https://github.com/di/calver/issues/20
+    monkeypatch.setattr(
+        calver.integration, "get_pkginfo_contents", pretend.raiser(FileNotFoundError)
+    )
+
+
+@pytest.fixture
+def source_date_epoch(monkeypatch):
+    _source_date_epoch = 1234567890
+    monkeypatch.setenv("SOURCE_DATE_EPOCH", str(_source_date_epoch))
+    yield _source_date_epoch
+    monkeypatch.delenv("SOURCE_DATE_EPOCH")

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -6,6 +6,12 @@ import pytest
 import calver.integration
 
 
+# Clean environment variable SOURCE_DATE_EPOCH if it's already present, e.g.
+# set up by packaging tool, since it may mess up logic of our testsuite.
+def pytest_configure(config):
+    if "SOURCE_DATE_EPOCH" in os.environ:
+        del os.environ["SOURCE_DATE_EPOCH"]
+
 @pytest.fixture
 def original_version():
     return pretend.stub()

--- a/tests/test_integration.py
+++ b/tests/test_integration.py
@@ -1,42 +1,10 @@
 import datetime
+import os
 
 import pretend
 import pytest
 
 import calver.integration
-
-
-@pytest.fixture
-def original_version():
-    return pretend.stub()
-
-
-@pytest.fixture
-def dist(original_version):
-    return pretend.stub(metadata=pretend.stub(version=original_version))
-
-
-@pytest.fixture
-def keyword():
-    return pretend.stub()
-
-
-@pytest.fixture
-def ignore_pkginfo(monkeypatch):
-    # Ensure that the test doesn't unintently prefer to read from a PKG_INFO
-    # that might exist in the source directory, e.g. when testing a sdist
-    # https://github.com/di/calver/issues/20
-    monkeypatch.setattr(
-        calver.integration, "get_pkginfo_contents", pretend.raiser(FileNotFoundError)
-    )
-
-
-@pytest.fixture
-def source_date_epoch(monkeypatch):
-    _source_date_epoch = 1234567890
-    monkeypatch.setenv("SOURCE_DATE_EPOCH", str(_source_date_epoch))
-    yield _source_date_epoch
-    monkeypatch.delenv("SOURCE_DATE_EPOCH")
 
 
 @pytest.mark.parametrize("value", [None, False, ""])

